### PR TITLE
assignment_2

### PR DESCRIPTION
**Pull request by jimmyabou**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

implementing and refining a machine learning pipeline using scikit-learn, focusing on preprocessing, model training, and evaluation

## What did you learn from the changes you have made?

how to create a pipeline to preprocess data, applying imputers, scaling and OneHotEncoder for numerical and categorical features respectively, and included a RandomForestClassifier in the pipeline for training and validation using k-fold cross-validation. Also learned how to extract validation metrics from the cross-validation process and finally how to evaluate the final model performance on the test set using the same metrics.

## How were these changes tested?

I used cross-validation with different evaluation metrics (neg_log_loss, ROC AUC, accuracy, balanced accuracy) to assess model performance

## Checklist
- [x ] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by ananyajha2000 — COMMENTED** [review]
A2: For the categorical transform:
One hot encoding strategy must be "infrequent_if_exists"

**Comment by ananyajha2000** [comment]
Please resubmit marked as incomplete

**Comment by jimmyabou** [comment]
> Please resubmit marked as incomplete

@ananyajha2000  can you please recheck the assignment?

**Review by ananyajha2000 — COMMENTED** [review]
A2 marked complete
